src/lib.rs

```diff
@@ -58,51 +57,64 @@ fn _logos_display(input: TokenStream2, debug: bool) -> TokenStream2 {
         _ => Err(syn::Error::new(span, "Can only derive display for enums"))
     };
     let traitt = if debug {
-        quote!(std::fmt::Debug)
+        quote!(core::fmt::Debug)
     } else {
-        quote!(std::fmt::Display)
+        quote!(core::fmt::Display)
     };
+    let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
     match resp {
         Ok(stream) => {
             quote! {
-                impl #traitt for #ident {
-                    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-                        let ret = match &self {
+                impl #impl_generics #traitt for #ident #ty_generics #where_clause {
+                    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                        use core::fmt::Write;
+                        match &self {
                             #stream
-                        };
-                        write!(f, "{}", ret)
+                        }
                     }
                 }
             }
-        },
+        }
         Err(e) => e.to_compile_error()
     }
 }
 
-fn gen_anon_args(n: usize) -> TokenStream2 {
+fn gen_anon_args(n: usize) -> Vec<proc_macro2::TokenStream> {
```

Comment:
Small nitpick. Could you change this type to `Vec<TokenStream2>` so that it is consistent with the rest of the crate?

Reply:
Sure thing, I'll also run `cargo fmt` since I forgot earlier.